### ISSUE TITLE
Update release documentation to match 16.1 process

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8271]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8271|
|Type|enhancement|
|Created|2016-04-04 17:33:14Z|
|Branch|https://github.com/twisted/twisted/tree/release-docs-8271|

Things I did differently:

- Gave ```--formats=bztar``` to sdist because the upload scripts didn't like it -- but we save a reasonable amount of space, so keeping bz2 is good

- used twine to upload tarball + whl

- there's also a bunch of broken links

- run the tests before review, get the build wheels then, not after its reviewed

<details><summary>Searchable metadata</summary>

```
trac-id__8271 8271
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Twisted_16_3_1 Twisted 16.3.1
branch__release_docs_8271 release-docs-8271
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1459791194967305 1459791194967305
changetime__1468433335238772 1468433335238772
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Ticket retargeted after milestone closed

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

I'm happy with this, please review.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc set owner to @hawkowl|
|-|-|

Looks OK.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub set status to `closed`|
|-|-|

In changeset 0d992a4f259ad90791c63be0ace04ceb79f63a37
```
#!CommitTicketReference repository="" revision="0d992a4f259ad90791c63be0ace04ceb79f63a37"
Merge pull request #327 from twisted/release-docs-8271: Update the release docs to meet 16.2/16.3

Author: hawkowl
Reviewer: rodrigc
Fixes: #8271
```
